### ISSUE TITLE
import WMS/WMTS  improve attribution

### ISSUE THREAD

**Issue by davidoesch** [geoadmin/mf-geoadmin3#3864]
Situation:
currently attribution is domain

Task
check if GetCap has
<Fees> and <AccessConstraints> at service level (<Service>…</Service)  or <Attribution> at the layer level (<Capability> -> <Layer> -> <Attribution>) According to 7.2.4.3 General service metadata  eg http://www.opengeospatial.org/standards/wms 
if those kewords exist

Action
-[ ] use Fees/Access constraints for footer copyright text
-[ ] if <Fees> and <AccessConstraints> Text is longer than 50 char, cut it at 50 caht
-[ ] if no  Fees/Access constraints is available or is "none" use domain name as we do for now

result
Attribution is optimized ( and no, I don't have a working exmample)

**Comment by p1d1d1** [comment]
This is a little tricky.
I try to summarize:

*WMS*
<Fees> and <AccessConstraints> are defined at service level and it's free text. Here is where one could say "Le service est réservé aux membres de ......."
<Attribution> is defined at layer level. Here is where one could state "Data from (c) OSM".

*WMTS*
<Attribution> at layer level in not available.

<Attribution> is IMHO more close to what we intend with attribution in map.geo.admin.ch. 
Unfortunately:
1) the tag is very rarely used
2) not available in WMTS

**Comment by davidoesch** [comment]
We focus on Service level-- here it is available for both WMS and WMTS

**Comment by sfkeller** [comment]
I support this issue to use 'Fee' at Service level.

I've investigated a bit: 7.2.4.3 General service metadata obviously does to ISO 19128 WMS spec. (not to WMTS - but Ok). Fee / AccessConstraints (e.g. with value 'none' when there s none) are also mentioned by INSPIRE Viewer Services and GDI-DE. There this seems to be a very important information in order to be compliant with their guidelines.

I currently don't know of any Swiss WMTS service - except german ones (see below). Given the GeoAdmin WMTS overlay feature is so new it's obvious that it's not yet there. 

As already mentioned, showing Attribution of an overlay is compliancy requirement of most geodata sources, including gov. data, commercial data and OSM. 

So, I think many WMTS providers would be happy when you raise the prio a little bit in order to make GeoAdmin an even greater service :-).

**Comment by davidoesch** [comment]
wms with access constraints:
https://owsproxy.lgl-bw.de/owsproxy/ows/WMS_Maps4BW?request=GetCapabilities&service=WMS&Version=1.3.0

**Comment by oterral** [comment]
Fee/Access constraint are always more than 50 charachters so it's useless to display it.

**Comment by sfkeller** [comment]
> Fee/Access constraint are always more than 50 characters so it's useless to display it.

That statement is not true. See e.g. (c) "OpenStreetMap Contributors" at osm.org or "© Mapbox © OpenStreetMap" at mapbox.com/maps which is around 24 chars.

There may be practical restrictions of max. char length to display. But let me be clear on this issue: The display of attribution is a legal requirement (and also an ethical) of any web app (including this one), and I since we have a solution at hand, there is no reason to wait implementing it now.

P.S. The WMTS from osm.ch is no ready to test the fees tag.

**Comment by oterral** [comment]
Yes for fee. I meant AccesConstraint is often more than 50 character.

**Comment by davidoesch** [comment]
As summarized by https://github.com/geoadmin/mf-geoadmin3/issues/3864#issuecomment-309389156 and defiend in the ticket: we need a solution for WMS,WMTS,KML,GPX,JSON --with fallback

**Comment by sfkeller** [comment]
Let's concentrate on services first, which is WMS,WMTS, and advance step by step.

**Comment by davidoesch** [comment]
we analyze how many of the current external sources support this feature https://github.com/geoadmin/mf-geoadmin3/blob/master/src/js/ImportController.js  then we decide how we proceed. and since import module is a joint dev with cantons and cities https://github.com/camptocamp/ngeo we will coordinate the approach with them as well ...

**Comment by boecklic** [comment]
Proposed solution: attribution line contains domain of wms as it is doing currently, if AccessConstraints available in wms/wmts getcap: show content of it (e.g. from https://owsproxy.lgl-bw.de/owsproxy/ows/WMS_Maps4BW?request=GetCapabilities&service=WMS&Version=1.3.0)
```
<AccessConstraints>
Der Darstellungsdienst Maps4BW kann für kommerzielle und nicht-kommerzielle Anwendungen unter den Bedingungen der Lizenz CC BY 3.0 (http://creativecommons.org/licenses/by/3.0/) kostenfrei genutzt werden. Die kostenfreie Nutzung von Maps4BW ist erlaubt, wenn an gut sichtbarer Stelle folgende Quellenangabe angeben wird: "Datengrundlage: LGL, www.lgl-bw". Maps4BW liegen nicht offene Geobasisdaten zugrunde, deren Nutzung einer gesonderten Vereinbarung bedarf.
</AccessConstraints>
```
it is shown in the popover/mouseover instead of the current "Dataset and/or style provided by third party"

**Comment by boecklic** [comment]
@boecklic commented: This is a little tricky.
I try to summarize:

*WMS*
<Fees> and <AccessConstraints> are defined at service level and it's free text. Here is where one could say "Le service est réservé aux membres de ......."
<Attribution> is defined at layer level. Here is where one could state "Data from (c) OSM".

*WMTS*
<Attribution> at layer level in not available.

<Attribution> is IMHO more close to what we intend with attribution in map.geo.admin.ch. 
Unfortunately:
1) the tag is very rarely used
2) not available in WMTS

**Comment by boecklic** [comment]
@boecklic commented: We focus on Service level-- here it is available for both WMS and WMTS

**Comment by boecklic** [comment]
@boecklic commented: I support this issue to use 'Fee' at Service level.

I've investigated a bit: 7.2.4.3 General service metadata obviously does to ISO 19128 WMS spec. (not to WMTS - but Ok). Fee / AccessConstraints (e.g. with value 'none' when there s none) are also mentioned by INSPIRE Viewer Services and GDI-DE. There this seems to be a very important information in order to be compliant with their guidelines.

I currently don't know of any Swiss WMTS service - except german ones (see below). Given the GeoAdmin WMTS overlay feature is so new it's obvious that it's not yet there. 

As already mentioned, showing Attribution of an overlay is compliancy requirement of most geodata sources, including gov. data, commercial data and OSM. 

So, I think many WMTS providers would be happy when you raise the prio a little bit in order to make GeoAdmin an even greater service :-).

**Comment by boecklic** [comment]
@boecklic commented: wms with access constraints:
https://owsproxy.lgl-bw.de/owsproxy/ows/WMS_Maps4BW?request=GetCapabilities&service=WMS&Version=1.3.0

**Comment by boecklic** [comment]
@boecklic commented: Fee/Access constraint are always more than 50 characters so it's useless to display it.

**Comment by boecklic** [comment]
@boecklic commented: > Fee/Access constraint are always more than 50 characters so it's useless to display it.

That statement is not true. See e.g. (c) "OpenStreetMap Contributors" at osm.org or "© Mapbox © OpenStreetMap" at mapbox.com/maps which is around 24 chars.

There may be practical restrictions of max. char length to display. But let me be clear on this issue: The display of attribution is a legal requirement (and also an ethical) of any web app including this one, and since we have a solution at hand, there is no reason to wait implementing it now.

P.S. The WMTS from osm.ch is now ready to test the fees tag.

**Comment by boecklic** [comment]
@boecklic commented: Yes for fee. I meant AccesConstraint is often more than 50 character.

**Comment by boecklic** [comment]
@boecklic commented: As summarized by https://github.com/geoadmin/mf-geoadmin3/issues/3864#issuecomment-309389156 and defiend in the ticket: we need a solution for WMS,WMTS,KML,GPX,JSON --with fallback

**Comment by boecklic** [comment]
@boecklic commented: Let's concentrate on services first, which is WMS,WMTS, and advance step by step.

**Comment by boecklic** [comment]
@boecklic commented: we analyze how many of the current external sources support this feature https://github.com/geoadmin/mf-geoadmin3/blob/master/src/js/ImportController.js  then we decide how we proceed. and since import module is a joint dev with cantons and cities https://github.com/camptocamp/ngeo we will coordinate the approach with them as well ...

**Comment by boecklic** [comment]
@boecklic commented: Proposed solution: attribution line contains domain of wms as it is doing currently, if AccessConstraints available in wms/wmts getcap: show content of it (e.g. from https://owsproxy.lgl-bw.de/owsproxy/ows/WMS_Maps4BW?request=GetCapabilities&service=WMS&Version=1.3.0)
```
<AccessConstraints>
Der Darstellungsdienst Maps4BW kann für kommerzielle und nicht-kommerzielle Anwendungen unter den Bedingungen der Lizenz CC BY 3.0 (http://creativecommons.org/licenses/by/3.0/) kostenfrei genutzt werden. Die kostenfreie Nutzung von Maps4BW ist erlaubt, wenn an gut sichtbarer Stelle folgende Quellenangabe angeben wird: "Datengrundlage: LGL, www.lgl-bw". Maps4BW liegen nicht offene Geobasisdaten zugrunde, deren Nutzung einer gesonderten Vereinbarung bedarf.
</AccessConstraints>
```
it is shown in the popover/mouseover instead of the current "Dataset and/or style provided by third party"
